src/many/src/server/module/_8_async.rs

```diff
@@ -0,0 +1,199 @@
+        e.bytes(&self.0)?;
+        Ok(())
+    }
+}
+
+impl<'b> Decode<'b> for AsyncToken {
+    fn decode(d: &mut Decoder<'b>) -> Result<Self, minicbor::decode::Error> {
+        Ok(Self(d.bytes()?.to_vec()))
+    }
+}
+
+impl From<Vec<u8>> for AsyncToken {
+    fn from(v: Vec<u8>) -> Self {
+        Self(v)
+    }
+}
+
+impl std::fmt::Debug for AsyncToken {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("AsyncToken")
+            .field(&hex::encode(&self.0))
+            .finish()
+    }
+}
+
+pub mod attributes {
+    use super::AsyncToken;
+    use crate::cbor::CborAny;
+    use crate::protocol::attributes::TryFromAttributeSet;
+    use crate::protocol::{Attribute, AttributeSet};
+    use crate::ManyError;
+
+    pub const ASYNC: Attribute = Attribute::id(1);
+
+    pub struct AsyncAttribute {
+        pub token: AsyncToken,
+    }
+
+    impl AsyncAttribute {
+        pub fn new(token: AsyncToken) -> Self {
+            Self { token }
+        }
+    }
+
+    impl From<AsyncAttribute> for Attribute {
+        fn from(a: AsyncAttribute) -> Attribute {
+            ASYNC.with_argument(CborAny::Bytes(a.token.0))
+        }
+    }
+
+    impl TryFrom<Attribute> for AsyncAttribute {
+        type Error = ManyError;
+
+        fn try_from(value: Attribute) -> Result<Self, Self::Error> {
+            if value.id != ASYNC.id {
+                return Err(ManyError::invalid_attribute_id(value.id.to_string()));
+            }
+
+            let arguments = value.into_arguments();
+            if arguments.len() != 1 {
+                Err(ManyError::invalid_attribute_arguments())
+            } else {
+                match arguments.into_iter().next() {
+                    Some(CborAny::Bytes(token)) => Ok(Self {
+                        token: token.into(),
+                    }),
+                    _ => Err(ManyError::invalid_attribute_arguments()),
+                }
+            }
+        }
+    }
+
+    impl TryFromAttributeSet for AsyncAttribute {
+        fn try_from_set(set: &AttributeSet) -> Result<Self, ManyError> {
+            match set.get_attribute(ASYNC.id) {
+                Some(attr) => AsyncAttribute::try_from(attr.clone()),
+                None => Err(ManyError::attribute_not_found(ASYNC.id.to_string())),
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, Encode, Decode)]
+#[cbor(map)]
+pub struct StatusArgs {
+    #[n(0)]
+    pub token: AsyncToken,
+}
+
+pub enum RequestStatus {
+    Unknown = 0,
+    Processing = 1,
+    Done = 2,
+}
+
+#[derive(Debug, Clone)]
+pub enum StatusReturn {
+    Unknown,
+    Queued,
+    Processing,
+    Done { response: CoseSign1 },
+    Expired,
+}
+
+impl Encode for StatusReturn {
+    fn encode<W: Write>(&self, e: &mut Encoder<W>) -> Result<(), Error<W::Error>> {
+        match self {
+            StatusReturn::Unknown => e.map(1)?.u8(0)?.u8(0)?,
```

Comment:
Use an `enum` value instead of hardcoding the return value ID?

Reply:
Can't use enum values with variants. But I isolated getting the variant id in a separate private function.

Reply:
Would it make sense to use something like https://crates.io/crates/num_enum? Maybe it's overkill, idk.